src/core/server/elasticsearch/elasticsearch_service.ts

```diff
@@ -154,10 +154,10 @@ export class ElasticsearchService
 
   private createClusterClient(
     type: string,
-    baseConfig: ElasticsearchConfig,
+    baseConfig: ElasticsearchClientConfig,
     clientConfig?: Partial<ElasticsearchClientConfig>
   ) {
-    const config = clientConfig ? merge({}, baseConfig, clientConfig) : baseConfig;
+    const config = clientConfig ? mergeConfig(baseConfig, clientConfig) : baseConfig;
```

Comment:
isnt this easier to read?

```
clientConfig: Partial<ElasticsearchClientConfig> = {}
) {

const config = mergeConfig(baseConfig, clientConfig);
```


I am also concerned that not using `mergeConfig` in both cases might cause unexpected bugs since the function `mergeConfig` has special logic that will only apply to 1 branch of this statement